### PR TITLE
start killing PlatformDefs

### PR DESCRIPTION
**Pull request by FernetMenta**
I was looing into #12342 and found that it failed on OSX because of this PlatformDefs crap. This PR starts with dropping it.

@pkerling #12342 builds on top of this.

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
@Memphiz this makes the XBMC_BOOL hack obsolete

**Comment by Memphiz** [comment]
Nice! This hack was annoying for such a long time. But first I have read "start killing platform devs" and wondered what kind of commits would achieve that ;) - I will try it out this evening and report back.

**Comment by Memphiz** [comment]
I compiled for ios successfully - on osx when running the ALL_BUILD target from xcode cmake is run everytime and cancels the build afterwards because it regenerates the xcode project each time. I know this is not related to this PR for sure. But it prevents me from test compiling it currently. Do you see the same behavior with the Xcode generator?

**Comment by Memphiz** [comment]
tried with makefiles - but the same issue is there too - whenever i hit make - cmake regenerates ... so it thinks something is dirty everytime. Also compile didn't get through - well out of time. Seems jenkins had no issue so its most likely ok - but this cmake problem will block all xcode generator users.

**Review by Memphiz — APPROVED** [review]
I guess its working ;)

**Review by Paxxi — CHANGES_REQUESTED** [review]
changing `BOOL` to `bool` is not equivalent and will cause compiler warnings. This requires changes to methods as well to return proper `true`/`false` instead of `TRUE`/`FALSE`

Also there's a whole lot of random whitespace changes. I don't mind really but people tend to complain about these.

**Comment by FernetMenta** [comment]
i changed BOOL to int where appropriate. please comment on the line

File: xbmc/linux/PlatformDefs.h [hunk 1]
```diff
@@ -188,7 +188,6 @@ typedef unsigned short  WORD;
 typedef unsigned char   BYTE;
 typedef char        CHAR;
 typedef wchar_t     WCHAR;
-typedef int         BOOL;
```
Comment (Rechi) [inline xbmc/linux/PlatformDefs.h]: this breaks TexturePacker at least on OSX.

**Review by Rechi — CHANGES_REQUESTED** [review]
TexturePacker needs to be fixed

File: xbmc/filesystem/iso9660.cpp [hunk 2]
```diff
@@ -180,7 +180,7 @@ struct iso_dirtree *iso9660::ReadRecursiveDirFromSector( DWORD sector, const cha
   if (!pCurr_dir_cache )
     return NULL;
 
-  BOOL bResult = ::ReadFile( m_info.ISO_HANDLE, pCurr_dir_cache, wSectorSize, &lpNumberOfBytesRead, NULL );
+  bool bResult = ::ReadFile( m_info.ISO_HANDLE, pCurr_dir_cache, wSectorSize, &lpNumberOfBytesRead, NULL );
```
Comment (Paxxi) [inline xbmc/filesystem/iso9660.cpp]: This will trigger performance warning, converting int to bool in MSVC

File: xbmc/filesystem/iso9660.cpp [hunk 3]
```diff
@@ -498,7 +498,7 @@ void iso9660::Scan()
     DWORD lpNumberOfBytesRead;
     char* pCurr_dir_cache = (char*)malloc( 16*wSectorSize );
     iso9660_Directory isodir;
-    BOOL bResult = ::ReadFile( m_info.ISO_HANDLE, pCurr_dir_cache, wSectorSize, &lpNumberOfBytesRead, NULL );
+    bool bResult = ::ReadFile( m_info.ISO_HANDLE, pCurr_dir_cache, wSectorSize, &lpNumberOfBytesRead, NULL );
```
Comment (Paxxi) [inline xbmc/filesystem/iso9660.cpp]: same here

File: xbmc/linux/XFileUtils.cpp [hunk 4]
```diff
@@ -239,7 +239,7 @@ DWORD  SetFilePointer(HANDLE hFile, int32_t lDistanceToMove,
 }
 
 // uses statfs
-BOOL GetDiskFreeSpaceEx(
+bool GetDiskFreeSpaceEx(
```
Comment (Paxxi) [inline xbmc/linux/XFileUtils.cpp]: This breaks the "emulation" as the method signature isn't the same, however this method is unused I think

File: xbmc/linux/XFileUtils.h [hunk 5]
```diff
@@ -28,19 +28,19 @@ HANDLE CreateFile(LPCTSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode,
             LPSECURITY_ATTRIBUTES lpSecurityAttributes,  DWORD dwCreationDisposition,
             DWORD dwFlagsAndAttributes, HANDLE hTemplateFile);
 
-BOOL   WriteFile(HANDLE hFile, const void * lpBuffer, DWORD nNumberOfBytesToWrite,  LPDWORD lpNumberOfBytesWritten, LPVOID lpOverlapped);
-BOOL   ReadFile( HANDLE hFile, LPVOID lpBuffer, DWORD nNumberOfBytesToRead, LPDWORD lpNumberOfBytesRead, void* unsupportedlpOverlapped);
+bool   WriteFile(HANDLE hFile, const void * lpBuffer, DWORD nNumberOfBytesToWrite,  LPDWORD lpNumberOfBytesWritten, LPVOID lpOverlapped);
```
Comment (Paxxi) [inline xbmc/linux/XFileUtils.h]: same here

File: xbmc/linux/XFileUtils.h [hunk 6]
```diff
@@ -28,19 +28,19 @@ HANDLE CreateFile(LPCTSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode,
...
+bool   ReadFile( HANDLE hFile, LPVOID lpBuffer, DWORD nNumberOfBytesToRead, LPDWORD lpNumberOfBytesRead, void* unsupportedlpOverlapped);
```
Comment (Paxxi) [inline xbmc/linux/XFileUtils.h]: same here

File: xbmc/linux/XFileUtils.h [hunk 7]
```diff
@@ -28,19 +28,19 @@ HANDLE CreateFile(LPCTSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode,
...
 
-DWORD  SetFilePointer(HANDLE hFile, int32_t lDistanceToMove,
+uint32_t  SetFilePointer(HANDLE hFile, int32_t lDistanceToMove,
                       int32_t *lpDistanceToMoveHigh, DWORD dwMoveMethod);
-BOOL   SetFilePointerEx(HANDLE hFile, LARGE_INTEGER liDistanceToMove,PLARGE_INTEGER lpNewFilePointer, DWORD dwMoveMethod);
+bool   SetFilePointerEx(HANDLE hFile, LARGE_INTEGER liDistanceToMove,PLARGE_INTEGER lpNewFilePointer, DWORD dwMoveMethod);
```
Comment (Paxxi) [inline xbmc/linux/XFileUtils.h]: same here

File: xbmc/linux/XTimeUtils.cpp [hunk 8]
```diff
@@ -77,7 +77,7 @@ VOID GetLocalTime(LPSYSTEMTIME sysTime)
   g_timezone.m_IsDST = now.tm_isdst;
 }
 
-BOOL FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileTime)
+bool FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileTime)
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: same here

File: xbmc/linux/XTimeUtils.cpp [hunk 9]
```diff
@@ -95,7 +95,7 @@ BOOL FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileT
   return true;
 }
 
-BOOL   SystemTimeToFileTime(const SYSTEMTIME* lpSystemTime,  LPFILETIME lpFileTime)
+bool   SystemTimeToFileTime(const SYSTEMTIME* lpSystemTime,  LPFILETIME lpFileTime)
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: and here

File: xbmc/linux/XTimeUtils.cpp [hunk 10]
```diff
@@ -155,7 +155,7 @@ LONG   CompareFileTime(const FILETIME* lpFileTime1, const FILETIME* lpFileTime2)
      return 1;
 }
 
-BOOL   FileTimeToSystemTime( const FILETIME* lpFileTime, LPSYSTEMTIME lpSystemTime)
+bool   FileTimeToSystemTime( const FILETIME* lpFileTime, LPSYSTEMTIME lpSystemTime)
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: here

File: xbmc/linux/XTimeUtils.cpp [hunk 11]
```diff
@@ -182,7 +182,7 @@ BOOL   FileTimeToSystemTime( const FILETIME* lpFileTime, LPSYSTEMTIME lpSystemTi
   return 1;
 }
 
-BOOL   LocalFileTimeToFileTime( const FILETIME* lpLocalFileTime, LPFILETIME lpFileTime)
+bool   LocalFileTimeToFileTime( const FILETIME* lpLocalFileTime, LPFILETIME lpFileTime)
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: here

File: xbmc/linux/XTimeUtils.cpp [hunk 12]
```diff
@@ -196,7 +196,7 @@ BOOL   LocalFileTimeToFileTime( const FILETIME* lpLocalFileTime, LPFILETIME lpFi
   return 1;
 }
 
-BOOL  FileTimeToTimeT(const FILETIME* lpLocalFileTime, time_t *pTimeT) {
+bool  FileTimeToTimeT(const FILETIME* lpLocalFileTime, time_t *pTimeT) {
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: here

File: xbmc/linux/XTimeUtils.cpp [hunk 13]
```diff
@@ -218,7 +218,7 @@ BOOL  FileTimeToTimeT(const FILETIME* lpLocalFileTime, time_t *pTimeT) {
   return true;
 }
 
-BOOL  TimeTToFileTime(time_t timeT, FILETIME* lpLocalFileTime) {
+bool  TimeTToFileTime(time_t timeT, FILETIME* lpLocalFileTime) {
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.cpp]: here

File: xbmc/linux/XTimeUtils.h [hunk 14]
```diff
@@ -22,17 +22,17 @@
...
-void WINAPI Sleep(DWORD dwMilliSeconds);
+void WINAPI Sleep(uint32_t dwMilliSeconds);
 
-BOOL   FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileTime);
-BOOL   SystemTimeToFileTime(const SYSTEMTIME* lpSystemTime,  LPFILETIME lpFileTime);
-LONG   CompareFileTime(const FILETIME* lpFileTime1, const FILETIME* lpFileTime2);
-BOOL   FileTimeToSystemTime( const FILETIME* lpFileTime, LPSYSTEMTIME lpSystemTime);
-BOOL   LocalFileTimeToFileTime( const FILETIME* lpLocalFileTime, LPFILETIME lpFileTime);
-VOID   GetSystemTimeAsFileTime(LPFILETIME lpSystemTimeAsFileTime);
+bool   FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileTime);
```
Comment (Paxxi) [inline xbmc/linux/XTimeUtils.h]: all of these

File: xbmc/linux/XFileUtils.cpp (hunk 4 above)
Comment (FernetMenta) [inline xbmc/linux/XFileUtils.cpp]: right, I will change former BOOL to int. thanks

File: xbmc/linux/PlatformDefs.h (hunk 1 above)
Comment (FernetMenta) [inline xbmc/linux/PlatformDefs.h]: how? I tested it on osx

Comment (Rechi) [inline xbmc/linux/PlatformDefs.h]: have you also completely rebuild depends?
https://github.com/xbmc/xbmc/blob/0f3ae09/tools/depends/native/TexturePacker/src/xwinapi.h#L28 uses `BOOL` from this file.

Comment (FernetMenta) [inline xbmc/linux/PlatformDefs.h]: ahh depends, will fix. thanks!

File: xbmc/linux/XTimeUtils.cpp [hunk 15]
```diff
@@ -92,10 +92,10 @@ BOOL FileTimeToLocalFileTime(const FILETIME* lpFileTime, LPFILETIME lpLocalFileT
 
   lpLocalFileTime->dwLowDateTime = l.u.LowPart;
   lpLocalFileTime->dwHighDateTime = l.u.HighPart;
-  return true;
```
Comment (FernetMenta) [inline xbmc/linux/XTimeUtils.cpp]: now it even fixes some BOOL bool mismatch we had before

**Comment by FernetMenta** [comment]
updated
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Review by Rechi — APPROVED** [review]

**Comment by FernetMenta** [comment]
@Paxxi I updated the PR. please review if your request was fully addressed.

**Review by Paxxi — APPROVED** [review]
looks good to me now.

Why also not change the WORD and DWORD in the waveformat struct? Good to merge but just curious.

**Comment by FernetMenta** [comment]
> Why also not change the WORD and DWORD in the waveformat struct?

Good question, this did not come to my mind. Probably because I was wondering why this struct is used by OMXAudio. @popcornmix this is now local to pi.

**Comment by MartijnKaijser** [comment]
@Montellese wasn't there some issue you had regarding BOOL that was defined in some way? Can;t recall what PR that was. Maybe it's now solved?

File: xbmc/cores/ExternalPlayer/ExternalPlayer.cpp [hunk 16]
```diff
@@ -452,13 +452,12 @@ BOOL CExternalPlayer::ExecuteAppW32(const char* strPath, const char* strSwitches
     CloseHandle(m_processInfo.hProcess);
     m_processInfo.hProcess = 0;
   }
-
-  return ret;
+  return (ret == 0);
```
Comment (Rechi) [inline xbmc/cores/ExternalPlayer/ExternalPlayer.cpp]: This changes behaviour.
After replacing it with `return ret == TRUE;` it fixes https://trac.kodi.tv/ticket/17983

File: xbmc/cores/ExternalPlayer/ExternalPlayer.cpp [hunk 17]
```diff
@@ -491,7 +490,7 @@ BOOL CExternalPlayer::ExecuteAppAndroid(const char* strSwitches,const char* strP
     CLog::Log(LOGNOTICE, "%s: Failure", __FUNCTION__);
   }
 
-  return ret;
+  return (ret == 0);
```
Comment (Rechi) [inline xbmc/cores/ExternalPlayer/ExternalPlayer.cpp]: `ret` is already a boolean and not `BOOL`

File: xbmc/cores/ExternalPlayer/ExternalPlayer.cpp (hunk 16 above)
Comment (FernetMenta) [reply xbmc/cores/ExternalPlayer/ExternalPlayer.cpp]: great, you found the root cause

**Pull request merged**